linera-service/src/node_service.rs

```diff
@@ -643,13 +728,17 @@ where
         &self,
         application_id: UserApplicationId,
         request: &Request,
+        chain_id: ChainId,
     ) -> Result<async_graphql::Response, NodeServiceError> {
         let bytes = serde_json::to_vec(&request)?;
         let query = Query::User {
             application_id,
             bytes,
         };
-        let response = self.client.lock().await.query_application(&query).await?;
+        let Some(client) = self.clients.client_lock(&chain_id).await else {
+            return Err(NodeServiceError::UnknownChainId);
+        };
```

Comment:
Shouldn't this be called `client_lock`? Then we remove `try_client_lock` altogether and use `client_lock(..)?` instead.

Reply:
We can do that, but then in most call sites we'll replace the `NodeServiceError` with a GraphQL error.

Alternatively we could have separate `client_lock_graphql` and `client_lock_node_service` functions, that just differ in the error type they return?